### ISSUE TITLE
 RUSTSEC-2022-0061: Crate `parity-wasm` deprecated by the author #2921 

### ISSUE THREAD

**Issue by kayabaNerve** [paritytech/substrate#13141]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

The deprecated parity-wasm crate is in use by sp_version, sc_executor, and the contracts pallet. Are there any migration efforts planned by Parity?

**Comment by bkchr** [comment]
Cc @athei

**Comment by athei** [comment]
I was the one who deprecated it because we don't want to maintain our own parser for wasm long term. It is to prevent new users to depend on it. Security issues and bugs will still be fixed if they are relevant to substrate.

We still depend on it because it is used by `wash-instrument` to inject stack height metering for the PvF. Eventually, we want to migrate to a combination of wasmparser and wasm-encode here. Additionally, pallet-contracts uses it for gas metering. However, we want to roll that into was to save the costly instrumentation step.

tl;dr It is okay to use it. Just don't add no new users unless absolutely necessary.
